clang-cl 11.0.0 linking error - duplicate symbols (Windows)

### Work environment

| Questions                                            | Answers
|------------------------------------------------------|--------------------
| OS/arch/bits (mandatory)                             | Windows x86_64
| File format of the file you reverse (mandatory)      | -
| Architecture/bits of the file (mandatory)            | x86/64
| r2 -v full output, **not truncated** (mandatory)     | https://github.com/radareorg/radare2/commit/14f82ee3012e10b56674a01c360ac1fbc4b4bf52

### Error

```
FAILED: libr/debug/r_debug.dll 
"lld-link"  /MACHINE:x64 /OUT:libr/debug/r_debug.dll libr/debug/r_debug.dll.p/arg.c.obj libr/debug/r_debug.dll.p/ddesc.c.obj libr/debug/r_debug.dll.p/debug.c.obj libr/debug/r_debug.dll.p/dreg.c.obj libr/debug/r_debug.dll.p/desil.c.obj libr/debug/r_debug.dll.p/dmap.c.obj libr/debug/r_debug.dll.p/pid.c.obj libr/debug/r_debug.dll.p/plugin.c.obj libr/debug/r_debug.dll.p/dsession.c.obj libr/debug/r_debug.dll.p/dsignal.c.obj libr/debug/r_debug.dll.p/snap.c.obj libr/debug/r_debug.dll.p/trace.c.obj libr/debug/r_debug.dll.p/p_bfvm.c.obj libr/debug/r_debug.dll.p/p_debug_bf.c.obj libr/debug/r_debug.dll.p/p_debug_bochs.c.obj libr/debug/r_debug.dll.p/p_debug_esil.c.obj libr/debug/r_debug.dll.p/p_debug_gdb.c.obj libr/debug/r_debug.dll.p/p_debug_io.c.obj libr/debug/r_debug.dll.p/p_debug_native.c.obj libr/debug/r_debug.dll.p/p_debug_null.c.obj libr/debug/r_debug.dll.p/p_debug_qnx.c.obj libr/debug/r_debug.dll.p/p_debug_rap.c.obj libr/debug/r_debug.dll.p/p_debug_winkd.c.obj libr/debug/r_debug.dll.p/p_debug_windbg.c.obj libr/debug/r_debug.dll.p/p_native_maps_windows_maps.c.obj libr/debug/r_debug.dll.p/p_native_windows_windows_debug.c.obj libr/debug/r_debug.dll.p/p_native_windows_windows_message.c.obj "/nologo" "/OPT:REF" "/DLL" "/IMPLIB:libr/debug\r_debug.lib" "/WHOLEARCHIVE:libr2sdb.a" "libr/util\r_util.lib" "libr2spp.a" "libr2sdb.a" "libr2zlib.a" "libr/hash\r_hash.lib" "libr/reg\r_reg.lib" "libr/syscall\r_syscall.lib" "libr/anal\r_anal.lib" "libr/flag\r_flag.lib" "libr/io\r_io.lib" "libr/bp\r_bp.lib" "libr/search\r_search.lib" "libr/cons\r_cons.lib" "libr/lang\r_lang.lib" "libr/egg\r_egg.lib" "shlr\libr2winkd.a" "libr/crypto\r_crypto.lib" "libr/socket\r_socket.lib" "shlr\libr2qnx.a" "shlr/gdb\libr2gdb.a" "shlr\libr2bochs.a" "shlr\w32dbg_wrap\libr2w32dbg_wrap.a" "ws2_32.lib" "wininet.lib" "psapi.lib" "ws2_32.lib" "wininet.lib" "psapi.lib" "ws2_32.lib" "wininet.lib" "psapi.lib" "ws2_32.lib" "wininet.lib" "psapi.lib" "ws2_32.lib" "wininet.lib" "psapi.lib" "kernel32.lib" "user32.lib" "gdi32.lib" "winspool.lib" "shell32.lib" "ole32.lib" "oleaut32.lib" "uuid.lib" "comdlg32.lib" "advapi32.lib"
lld-link: error: duplicate symbol: w32_CreateToolhelp32Snapshot
>>> defined at libr/debug/r_debug.dll.p/p_debug_native.c.obj
>>> defined at libr/debug/r_debug.dll.p/p_native_maps_windows_maps.c.obj
lld-link: error: duplicate symbol: w32_GetModuleBaseName
>>> defined at libr/debug/r_debug.dll.p/p_debug_native.c.obj
>>> defined at libr/debug/r_debug.dll.p/p_native_maps_windows_maps.c.obj
lld-link: error: duplicate symbol: w32_GetModuleInformation
>>> defined at libr/debug/r_debug.dll.p/p_debug_native.c.obj
>>> defined at libr/debug/r_debug.dll.p/p_native_maps_windows_maps.c.obj
lld-link: error: duplicate symbol: w32_DebugActiveProcessStop
>>> defined at libr/debug/r_debug.dll.p/p_debug_native.c.obj
>>> defined at libr/debug/r_debug.dll.p/p_native_maps_windows_maps.c.obj
lld-link: error: duplicate symbol: w32_OpenThread
>>> defined at libr/debug/r_debug.dll.p/p_debug_native.c.obj
>>> defined at libr/debug/r_debug.dll.p/p_native_maps_windows_maps.c.obj
lld-link: error: duplicate symbol: w32_DebugBreakProcess
>>> defined at libr/debug/r_debug.dll.p/p_debug_native.c.obj
>>> defined at libr/debug/r_debug.dll.p/p_native_maps_windows_maps.c.obj
lld-link: error: duplicate symbol: w32_GetThreadId
>>> defined at libr/debug/r_debug.dll.p/p_debug_native.c.obj
>>> defined at libr/debug/r_debug.dll.p/p_native_maps_windows_maps.c.obj
lld-link: error: duplicate symbol: w32_GetProcessId
>>> defined at libr/debug/r_debug.dll.p/p_debug_native.c.obj
>>> defined at libr/debug/r_debug.dll.p/p_native_maps_windows_maps.c.obj
lld-link: error: duplicate symbol: w32_OpenProcess
>>> defined at libr/debug/r_debug.dll.p/p_debug_native.c.obj
>>> defined at libr/debug/r_debug.dll.p/p_native_maps_windows_maps.c.obj
lld-link: error: duplicate symbol: w32_QueryFullProcessImageName
>>> defined at libr/debug/r_debug.dll.p/p_debug_native.c.obj
>>> defined at libr/debug/r_debug.dll.p/p_native_maps_windows_maps.c.obj
lld-link: error: duplicate symbol: w32_GetMappedFileName
>>> defined at libr/debug/r_debug.dll.p/p_debug_native.c.obj
>>> defined at libr/debug/r_debug.dll.p/p_native_maps_windows_maps.c.obj
lld-link: error: duplicate symbol: w32_NtQuerySystemInformation
>>> defined at libr/debug/r_debug.dll.p/p_debug_native.c.obj
>>> defined at libr/debug/r_debug.dll.p/p_native_maps_windows_maps.c.obj
lld-link: error: duplicate symbol: w32_NtQueryInformationThread
>>> defined at libr/debug/r_debug.dll.p/p_debug_native.c.obj
>>> defined at libr/debug/r_debug.dll.p/p_native_maps_windows_maps.c.obj
lld-link: error: duplicate symbol: w32_NtDuplicateObject
>>> defined at libr/debug/r_debug.dll.p/p_debug_native.c.obj
>>> defined at libr/debug/r_debug.dll.p/p_native_maps_windows_maps.c.obj
lld-link: error: duplicate symbol: w32_NtQueryObject
>>> defined at libr/debug/r_debug.dll.p/p_debug_native.c.obj
>>> defined at libr/debug/r_debug.dll.p/p_native_maps_windows_maps.c.obj
lld-link: error: duplicate symbol: w32_GetEnabledXStateFeatures
>>> defined at libr/debug/r_debug.dll.p/p_debug_native.c.obj
>>> defined at libr/debug/r_debug.dll.p/p_native_maps_windows_maps.c.obj
lld-link: error: duplicate symbol: w32_InitializeContext
>>> defined at libr/debug/r_debug.dll.p/p_debug_native.c.obj
>>> defined at libr/debug/r_debug.dll.p/p_native_maps_windows_maps.c.obj
lld-link: error: duplicate symbol: w32_GetXStateFeaturesMask
>>> defined at libr/debug/r_debug.dll.p/p_debug_native.c.obj
>>> defined at libr/debug/r_debug.dll.p/p_native_maps_windows_maps.c.obj
lld-link: error: duplicate symbol: w32_LocateXStateFeature
>>> defined at libr/debug/r_debug.dll.p/p_debug_native.c.obj
>>> defined at libr/debug/r_debug.dll.p/p_native_maps_windows_maps.c.obj
lld-link: error: duplicate symbol: w32_SetXStateFeaturesMask
>>> defined at libr/debug/r_debug.dll.p/p_debug_native.c.obj
>>> defined at libr/debug/r_debug.dll.p/p_native_maps_windows_maps.c.obj
lld-link: error: too many errors emitted, stopping now (use /errorlimit:0 to see all errors)
[1405/1546] Compiling C object libr/fs/r_fs.dll.p/p_fs_hfs.c.obj
[1406/1546] Compiling C object libr/fs/r_fs.dll.p/p_fs_ntfs.c.obj
ninja: build stopped: subcommand failed.
[r2-meson][ERROR]: Ninja error. Exiting.
```

--- a/libr/debug/p/native/windows/windows_debug.h
+++ b/libr/debug/p/native/windows/windows_debug.h
@@ -118,28 +118,28 @@ typedef struct{
 	char *Name;
 } LIB_ITEM, *PLIB_ITEM;
 
-DWORD (WINAPI *w32_GetModuleBaseName)(HANDLE, HMODULE, LPTSTR, DWORD);
-BOOL (WINAPI *w32_GetModuleInformation)(HANDLE, HMODULE, LPMODULEINFO, DWORD);
-BOOL (WINAPI *w32_DebugActiveProcessStop)(DWORD);
-HANDLE (WINAPI *w32_OpenThread)(DWORD, BOOL, DWORD);
-BOOL (WINAPI *w32_DebugBreakProcess)(HANDLE);
-DWORD (WINAPI *w32_GetThreadId)(HANDLE); // Vista
-DWORD (WINAPI *w32_GetProcessId)(HANDLE); // XP
-HANDLE (WINAPI *w32_OpenProcess)(DWORD, BOOL, DWORD);
-BOOL (WINAPI *w32_QueryFullProcessImageName)(HANDLE, DWORD, LPTSTR, PDWORD);
-DWORD (WINAPI *w32_GetMappedFileName)(HANDLE, LPVOID, LPTSTR, DWORD);
-NTSTATUS (WINAPI *w32_NtQuerySystemInformation)(ULONG, PVOID, ULONG, PULONG);
-NTSTATUS (WINAPI *w32_NtQueryInformationThread)(HANDLE, ULONG, PVOID, ULONG, PULONG);
-NTSTATUS (WINAPI *w32_NtDuplicateObject)(HANDLE, HANDLE, HANDLE, PHANDLE, ACCESS_MASK, ULONG, ULONG);
-NTSTATUS (WINAPI *w32_NtQueryObject)(HANDLE, ULONG, PVOID, ULONG, PULONG);
+static DWORD (WINAPI *w32_GetModuleBaseName)(HANDLE, HMODULE, LPTSTR, DWORD);
+static BOOL (WINAPI *w32_GetModuleInformation)(HANDLE, HMODULE, LPMODULEINFO, DWORD);
+static BOOL (WINAPI *w32_DebugActiveProcessStop)(DWORD);
+static HANDLE (WINAPI *w32_OpenThread)(DWORD, BOOL, DWORD);
+static BOOL (WINAPI *w32_DebugBreakProcess)(HANDLE);
+static DWORD (WINAPI *w32_GetThreadId)(HANDLE); // Vista
+static DWORD (WINAPI *w32_GetProcessId)(HANDLE); // XP
+static HANDLE (WINAPI *w32_OpenProcess)(DWORD, BOOL, DWORD);
+static BOOL (WINAPI *w32_QueryFullProcessImageName)(HANDLE, DWORD, LPTSTR, PDWORD);
+static DWORD (WINAPI *w32_GetMappedFileName)(HANDLE, LPVOID, LPTSTR, DWORD);
+static NTSTATUS (WINAPI *w32_NtQuerySystemInformation)(ULONG, PVOID, ULONG, PULONG);
+static NTSTATUS (WINAPI *w32_NtQueryInformationThread)(HANDLE, ULONG, PVOID, ULONG, PULONG);
+static NTSTATUS (WINAPI *w32_NtDuplicateObject)(HANDLE, HANDLE, HANDLE, PHANDLE, ACCESS_MASK, ULONG, ULONG);
+static NTSTATUS (WINAPI *w32_NtQueryObject)(HANDLE, ULONG, PVOID, ULONG, PULONG);
 // fpu access API
-ut64 (WINAPI *w32_GetEnabledXStateFeatures)();
-BOOL (WINAPI *w32_InitializeContext)(PVOID, DWORD, PCONTEXT*, PDWORD);
-BOOL (WINAPI *w32_GetXStateFeaturesMask)(PCONTEXT Context, PDWORD64);
-PVOID (WINAPI *w32_LocateXStateFeature)(PCONTEXT Context, DWORD, PDWORD);
-BOOL (WINAPI *w32_SetXStateFeaturesMask)(PCONTEXT Context, DWORD64);
-DWORD (WINAPI *w32_GetModuleFileNameEx)(HANDLE, HMODULE, LPTSTR, DWORD);
-HANDLE (WINAPI *w32_CreateToolhelp32Snapshot)(DWORD, DWORD);
+static ut64 (WINAPI *w32_GetEnabledXStateFeatures)();
+static BOOL (WINAPI *w32_InitializeContext)(PVOID, DWORD, PCONTEXT*, PDWORD);
+static BOOL (WINAPI *w32_GetXStateFeaturesMask)(PCONTEXT Context, PDWORD64);
+static PVOID (WINAPI *w32_LocateXStateFeature)(PCONTEXT Context, DWORD, PDWORD);
+static BOOL (WINAPI *w32_SetXStateFeaturesMask)(PCONTEXT Context, DWORD64);
+static DWORD (WINAPI *w32_GetModuleFileNameEx)(HANDLE, HMODULE, LPTSTR, DWORD);
+static HANDLE (WINAPI *w32_CreateToolhelp32Snapshot)(DWORD, DWORD);
 
 // APIs
 int w32_init(RDebug *dbg);

--- a/shlr/heap/include/r_windows/windows_heap.h
+++ b/shlr/heap/include/r_windows/windows_heap.h
@@ -988,22 +988,22 @@ typedef struct _HeapInformation {
 	DEBUG_HEAP_INFORMATION heaps[/* count */];
 } HeapInformation, *PHeapInformation;
 
-PDEBUG_BUFFER (NTAPI *RtlCreateQueryDebugBuffer)(
+static PDEBUG_BUFFER (NTAPI *RtlCreateQueryDebugBuffer)(
 	IN DWORD Size,
 	IN BOOLEAN EventPair
 );
 
-NTSTATUS (NTAPI *RtlQueryProcessDebugInformation)(
+static NTSTATUS (NTAPI *RtlQueryProcessDebugInformation)(
 	IN DWORD ProcessId,
 	IN DWORD DebugInfoClassMask,
 	IN OUT PDEBUG_BUFFER DebugBuffer
 );
 
-NTSTATUS (NTAPI *RtlDestroyQueryDebugBuffer)(
+static NTSTATUS (NTAPI *RtlDestroyQueryDebugBuffer)(
 	IN PDEBUG_BUFFER DebugBuffer
 );
 
-__kernel_entry NTSTATUS (NTAPI *w32_NtQueryInformationProcess)(
+static __kernel_entry NTSTATUS (NTAPI *w32_NtQueryInformationProcess)(
   IN HANDLE           ProcessHandle,
   IN PROCESSINFOCLASS ProcessInformationClass,
   OUT PVOID           ProcessInformation,

